e2etests/tests/sessions.go

```diff
@@ -111,60 +102,35 @@ var _ = Describe("Router Host configuration", Ordered, func() {
 			Expect(err).NotTo(HaveOccurred())
 		})
 
-		It("receives type 5 routes from the fabric", func() {
-			leafAConfig := infra.LeafConfiguration{
-				Leaf: infra.LeafAConfig,
-				Red: infra.Addresses{
-					IPV4: []string{"192.168.20.0/24"},
-				},
-			}
-
-			config, err := infra.LeafConfigToFRR(leafAConfig)
-			Expect(err).NotTo(HaveOccurred())
-
-			By("announcing type 5 routes from leafA")
-			err = infra.LeafAContainer.ReloadConfig(config)
-			Expect(err).NotTo(HaveOccurred())
-
-			Eventually(func() error {
-				for _, p := range routerPods {
-					exec := executor.ForPod(p.Namespace, p.Name, "frr")
-					evpn, err := frr.EVPNInfo(exec)
-					Expect(err).NotTo(HaveOccurred())
-					if !evpn.ContainsType5Route("192.168.20.0", leafAConfig.VTEPIP) {
-						return fmt.Errorf("type5 route for 192.168.20.0 - %s not found in %v in router %s", leafAConfig.VTEPIP, evpn, p.Name)
-					}
-				}
-				return nil
-			}, 2*time.Minute, time.Second).ShouldNot(HaveOccurred())
-
-		})
-
-		It("establishes a session with the host", func() {
+		It("establishes a session with the host and then removes it when deleting the vni", func() {
 			frrConfig, err := frrk8s.ConfigFromVNI(vni)
 			Expect(err).ToNot(HaveOccurred())
 			err = Updater.Update(config.Resources{
 				FRRConfigurations: []frrk8sv1beta1.FRRConfiguration{frrConfig},
 			})
 			Expect(err).NotTo(HaveOccurred())
 
-			ipToCheck, err := openperouter.RouterIPFromCIDR(vni.Spec.LocalCIDR)
+			validateFRRK8sSessionForVNI(vni, frrk8sPods, Established)
+
+			By("deleting the vni removes the session with the host")
+			err = Updater.Client().Delete(context.Background(), &vni)
 			Expect(err).NotTo(HaveOccurred())
 
-			for _, p := range frrk8sPods {
-				exec := executor.ForPod(p.Namespace, p.Name, "frr")
-				Eventually(func() error {
-					neigh, err := frr.NeighborInfo(ipToCheck, exec)
-					if err != nil {
-						return err
-					}
-					if neigh.BgpState != "Established" {
-						return fmt.Errorf("neighbor %s is not established", ipToCheck)
-					}
-					return nil
-				}, 2*time.Minute, time.Second).ShouldNot(HaveOccurred())
-			}
+			validateFRRK8sSessionForVNI(vni, frrk8sPods, !Established)
 		})
 
+		// This test must be the last of the ordered describe as it will remove the underlay
+		It("deleting the underlay removes the session with the tor", func() {
+			By("deleting the vni removes the session with the host")
```

Comment:
can/should we start with validating the session is there?

Reply:
I can, we already validated as the first it of the spec and this is an ordered spec, but re-validating won't hurt.